### PR TITLE
VideoPlayer: allow saving video buffers on flush

### PR DESCRIPTION
**Pull request by FernetMenta**
on events like DisplayLost video buffers can be saved, if implementation allows it.

fixes black screen when going full screen/windowed while video is paused

### CODE REVIEW

File: xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.h [hunk 1]
```diff
@@ -133,6 +133,7 @@ class CLinuxRendererGL : public CBaseRenderer
   virtual bool LoadShadersHook() { return false; };
   virtual bool RenderHook(int idx) { return false; };
   virtual void AfterRenderHook(int idx) {};
+  virtual bool CanSafeBuffers() { return true; };
```
Comment (garbear) [inline xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.h]: CanSaveBuffers()

Comment (FernetMenta) [reply xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.h]: thanks!

File: xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp [hunk 2]
```diff
@@ -452,17 +454,24 @@ void CLinuxRendererGL::LoadPlane(YUVPLANE& plane, int type,
     glBindBufferARB(GL_PIXEL_UNPACK_BUFFER_ARB, 0);
 }
 
-void CLinuxRendererGL::Flush()
+bool CLinuxRendererGL::Flush(bool saveBuffers)
 {
+  bool safe = saveBuffers && CanSaveBuffers();
```
Comment (ksooo) [inline xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp]: save, not safe;-)

**Review by ksooo — COMMENTED** [review]
safe vs save

File: xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp (hunk 2 above)
Comment (FernetMenta) [reply xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp]: here it is "safe" because I mean the state (!= verb)

Comment (ksooo) [reply xbmc/cores/VideoPlayer/VideoRenderers/LinuxRendererGL.cpp]: Ooops. Okay, then.

**Pull request merged**